platforms/software/dependency-management/src/main/java/org/gradle/internal/component/external/model/DefaultCapabilitiesMetadata.java

```diff
@@ -14,25 +14,27 @@
  * limitations under the License.
  */
 
-package org.gradle.api.internal.capabilities;
+package org.gradle.internal.component.external.model;
 
+import com.google.common.collect.ImmutableList;
 import org.gradle.api.capabilities.CapabilitiesMetadata;
 import org.gradle.api.capabilities.Capability;
 
 import java.util.List;
 
 /**
- * Internal extension of {@link CapabilitiesMetadata} that adds methods not intended for public consumption.
+ * An immutable implementation of {@link CapabilitiesMetadata}.
```

Comment:
Maybe worth calling out that this is specifically _shallowly_ immutable, as opposed to `ImmutableCapabilities` which is deep?

Reply:
Adjusted to wrap an `ImmutableCapabilities` and make this deeply immutable, as in practice it was always deeply immutable